Gives Radstorm a spitshine (new-rad wahoo)

TsunamiAnt opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

Adds alarm to radstorm

changes the visual to be more of a static-y buzz, bit subtler but still easily recognizable

makes the radstorm use the new fancy radiation system we added

removes dna and mutation mutations from it

I had to touch a thingy in weather-listener that I do not fully understand. Please take a look over it just for sanity's sake.

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

It has always looked like absolute garbage. If it was up to me i would have removed the visual effect entirely, but we're lucky it's not up to me, because I kinda like this more even.

The siren is also very vibey and not too high-key.

Additionally, using new radiation systems in places where they make sense to use is also very nice.

I fucking despise the dna mutation thing, it is so often irreversible and, for me and for some others, will basically just mean an instant round-end because we're gonna go fucking cryo/kill ourselves because we look like utter ass.

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding.
-->

## Testing Photographs and Procedure
<!-- Include any screenshots/videos/debugging steps of the modified code functioning successfully, ideally including edge cases. -->
<details>
<summary>Screenshots&Videos</summary>

https://github.com/user-attachments/assets/1e21f65c-f2b8-4c8b-81b7-646612d47614

</details>

## Changelog
:cl:
qol: added rad-damage to radstorms and removed their forced mutating
sound: radstorms now have a fancy siren
image: radstorms now have a fancy new visual effect
code: changed some code in weather-listener to account for off-lavaland weather soundloops
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

TsunamiAnt committed:
Updates radiation storm effects

Updates the radiation storm's visual and auditory feedback to be more thematically appropriate.

Changes the weather overlay to a radiation-specific effect.
Removes the unnecessary weather color property.

TsunamiAnt committed:
Adds rad storm weather event with sound alerts.

Adds a new radiation storm weather event. This includes new looping sounds to alert players inside and outside, differentiating between maintenance areas and other areas.
It also adds the weather listener element to possessed objects and characters on login, providing appropriate ambient sounds.
Radiation exposure is now handled through the radiation system.

PowerfulBacon reviewed (changes requested):
This is literally impossible to see

TsunamiAnt committed:
Updates radstorm weather effect icon.

Updates the weather_effects.dmi file to visually represent the radstorm effect. This change enhances the visual feedback for players experiencing the radstorm event.

PowerfulBacon reviewed (dismissed):
Needs to be greenish to look like radiation in the game. You could try the following things
- Just make the sprite green
- Make it green and apply some kind of emissive overlay too
- Set the luminosity of the area to 1 and also add the starlight overlay setting the colour to green

Right now it looks more like a slenderman/horror effect than radiation as our radiation doesn't look like camera static

TsunamiAnt committed:
Green

Improves the radiation storm effect by introducing a weather color.

This provides visual feedback to players, making it easier to identify
when a radiation storm is occurring.

TsunamiAnt committed:
Updates radstorm weather effect icon

Updates the weather effects DMI to visually represent the ongoing radstorm changes.

Rukofamicom reviewed (changes requested):
Ensure the false alarm radstorm is not invalidated by this.

TsunamiAnt committed:
We now have fake sound

Rukofamicom commented on code/datums/weather/weather_types/radiation_storm.dm:
```diff
@@ -35,51 +35,47 @@
-
-	if(prob(50))
-		do_mutate(human)
+	SSradiation.irradiate(human, intensity = rand(1, 5))
```
```suggestion
	SSradiation.irradiate(human, intensity = rand(5, 15))
```

I tested this in local with dummies and they were able to stand through the ENTIRE radstorm without difficulty

Rukofamicom reviewed (changes requested):
Disregard what I said in this message, I'll be adding this to radiation itself in another PR

Rukofamicom reviewed (approved):
Further personal testing shows that the radstorm not having substantial impact was more to do with how radiation works now. 

I rescind the final change request entirely, this is good to go imo.
